### ISSUE TITLE
Inconsistancy with parse_date_time(orders = 'YAU'), to obtain monday of a specific week.

### ISSUE THREAD

**Issue by demarsylvain** [tidyverse/lubridate#944]
I'm using the function `parse_date_time()` with the argument `orders = 'YAU'` in order to transform a year and week number into a Monday date. For instance, `'2017Monday1'` give `'2017-01-02'`, the first Monday of the year 2017. But for some years, like 2018, there is a gap of one week.

```r
parse_date_time('2017Monday1', orders = 'YAU')
#"2017-01-02 UTC"                # ok, first monday of 2017

parse_date_time('2018Monday1', orders = 'YAU')
#"2018-01-08 UTC"                # not ok, the first Monday of 2018 is 2018-01-01

### use 0-based for 2018
week_num <- 1 
parse_date_time(paste0('2018Monday', week_num - 1), orders = 'YAU')
#"2018-01-01 UTC"               # ok

### now, use 0-based for 2017
week_num <- 1
parse_date_time(paste0('2017Monday', week_num - 1), orders = 'YAU')
#"2017-01-02 UTC"                # ok

week_num <- 2
parse_date_time(paste0('2017Monday', 2 - 1), orders = 'YAU')
#"2017-01-02 UTC"                # not ok, as it's still the first week
```

For 2018, I need to used a 0-based operation to obtain the right mondays, but not for 2017. 
Is it normal ?

Thank you in advance.

**Comment by vspinu** [comment]
Lubridate relies on `strptime` for this. From the docs:

```
     ‘U’ Week of the year as decimal number (00-53 or 0-53) using
          Sunday as the first day 1 of the week (and typically with the
          first Sunday of the year as day 1 of week 1).  The US
          convention.
```

I think this means the following. Week 1 starts on first Sunday, everything before first Sunday is considered week 0. So for `2018` the week 1 is  Jan07--Jan13, thus `2018Monday01` is parsed as `2018-01-08`, and `2018Monday00` is parsed as `2018-01-01` because it's an incomplete week 0 which contains Monday. 

 For 2017 the week 1 is jan01-jan07 and the week 0 is empty and does not contain the Monday. Similarly for 2019. 

Basically the rule is, if week 0 does not contain Monday, we fall back on first Monday of the year. I cannot judge if it's a good rule or not. I think the assumption is that things like `2017Monday00` should not be there in the first place. For sure, `strftime` does not generate such strings. You probably have to check with the source of those strings if they are under your control.
